src/components/icon/__spec__.js

```diff
@@ -160,24 +162,21 @@ describe('Icon', () => {
   describe("tags", () => {
     describe("on component", () => {
       let wrapper = shallow(<Icon data-element='bar' data-role='baz' type='tick'/>);
-
       it('include correct component, element and role data tags', () => {
         rootTagTest(wrapper, 'icon', 'bar', 'baz');
       });
     });
 
     describe("on internal elements", () => {
-      let wrapper = shallow(
+      let wrapper = mount(
         <Icon
           tooltipMessage='Test'
           tooltipAlign='left'
           tooltipPosition='top'
           type='tick'
         />);
-
-      elementsTagTest(wrapper, [
-        'tooltip'
-      ]);
+      wrapper.setState({ isVisible: true });
+      expect(document.getElementsByTagName('body')[0].innerHTML).toEqual(expect.stringContaining('data-component="tooltip" data-element="tooltip"'));
```

Comment:
need to find a better way of testing this

Reply:
```js
const tooltip = Icon.find(Tooltip);
elementsTagTest(tooltip, [
  'tooltip'
]);
```
??

Reply:
This was the test I was talking earlier. Yes not good. Your solution looks like the way to go but, inside the function `elementsTagTest` tries `wrapper.find({ 'data-element': element }).length` and return ,'2' for the length so it fails. Any idea of why  it would do that?